Documentation/Design/Lambdas/Lambda Declarations.md

```diff
@@ -0,0 +1,150 @@
+# Lambda Declarations
+
+GLS will support declaring lambda functions. In some languages, in order to declare a function that takes a lambda function as a parameter, the lambda must first be declared. This typically involves declaring the return type as well as the parameter names and types, which allow for compile-time type checking when the lambda is used.
+
+## Commands
+
+### `lambda declare`
+`lambda declare : typeName functionName returnType *`[parameterName parameterType ...]`*
+
+## Usage:
+
+```gls
+lambda declare : overMinimum check boolean age int
+```
+
+### CSharp:
+```CSharp
+delegate bool overMinimum(int age);
+```
+
+### Java:  
+```Java
+interface overMinimum {
+    boolean check(int age);
+}
+```
+
+### Python:
+Python does not require declaring lambdas.
+
+### Ruby:
+Ruby does not require declaring lambdas.
+
+### TypeScript
+```TypeScript
+interface IOverMinimum {
+    (age: number): boolean;
+}
+```
+
+## Implementation
+
+### Properties
+
+Properties will be stored in the LambdaProperties object. The following properties will be added.
+
+<table>
+    <thead>
+        <th>Property Name</th>
+        <th>Type</th>
+        <th>Description</th>
+    </thead>
+    <tbody>
+        <tr>
+            <td>requiresDeclaration</td>
```

Comment:
I ordered these properties by type (booleans, then strings) and then in the order they're needed. Should we alphabetize instead?